### PR TITLE
EUI-1937-add-share-case-button-depends-on-EUI-1940

### PR DESCRIPTION
**Pull request by rayliangatos**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/cases/containers/case-list/case-list.component.ts [hunk 1]
```diff
@@ -59,6 +59,7 @@ export class CaseListComponent implements OnInit, OnDestroy {
   public paginationSubscription: Subscription;
   public isVisible: boolean;
   public jurisdictions: Jurisdiction[];
+  public selectedCases: SearchResultViewItem[] = [];
```
Comment (ronaldmansveld) [inline src/cases/containers/case-list/case-list.component.ts:62]: These should be stored in the store, as they're needed on the next page when the button is clicked.

Comment (rayliangatos) [reply src/cases/containers/case-list/case-list.component.ts:62]: the selected case is not added to store, please review.

File: src/cases/containers/case-list/case-list.component.spec.ts [hunk 2]
```diff
@@ -350,6 +350,49 @@ describe('CaseListComponent', () => {
...
+      expect(component.checkIfButtonDisabled()).toBeTruthy();
+    });
+    it('Share a case button is selectable when any case is selected', () => {
+      component.retrieveSelections(selectedCases);
+      expect(component.checkIfButtonDisabled()).toBeFalsy();
+    });
+    it('Should save share cases to store', () => {
+      component.retrieveSelections(selectedCases);
+      component.shareCaseSubmit();
+      expect(component.checkIfButtonDisabled()).toBeFalsy();
```
Comment (sreekanthpuligadda) [inline src/cases/containers/case-list/case-list.component.spec.ts:389]: Can we have validation to check cases saved to store.

Comment (rayliangatos) [reply src/cases/containers/case-list/case-list.component.spec.ts:389]: that has been in:
src/cases/store/selectors/share-case.selectors.spec.ts

Comment (sreekanthpuligadda) [reply src/cases/containers/case-list/case-list.component.spec.ts:389]: share-case.selectors.spec.ts is validating store specific functionality.

For this component method shareCaseSubmit(), good to have its own validation. As i see the test, if we comment out code from shareCaseSubmit() method this test would still pass defeating the purpose of this test as we not validating expected output from this method. Correct me if i am wrong.

Comment (rayliangatos) [reply src/cases/containers/case-list/case-list.component.spec.ts:389]: if you comment out the code from shareCaseSubmit() method in ts, the spec.ts will fail to compile, that is what the coverage should be.
if you comment out the  shareCaseSubmit() from html, yes correct the test will still be passing, but I'm not sure whether we need to have such a detailed and high rate coverage test on html as well. Hope that make sense.

Comment (sreekanthpuligadda) [reply src/cases/containers/case-list/case-list.component.spec.ts:389]: Not sure if the test that is passing is valid if method shareCaseSubmit() does nothing. See screenshot. 
There is lots of focus on unit tests coverage. please see unit tests section 
 https://tools.hmcts.net/confluence/display/RQA/QA+Review+-+ExUI . "Scope of tests will include valid, invalid & edge cases "

image

File: src/cases/store/reducers/share-case.reducer.spec.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        }
+      }];
+      const oldAction = new fromActions.AddShareCases(selectedCases);
+      const oldState = fromReducer.shareCasesReducer(initialState, oldAction);
+      const newAction = new fromActions.AddShareCases(addedSelectedCases);
+      const newState = fromReducer.shareCasesReducer(oldState, newAction);
+      expect(newState.shareCases.length).toEqual(3);
+    });
+
+    it('should delete a case from store', () => {
```
Comment (sreekanthpuligadda) [inline src/cases/store/reducers/share-case.reducer.spec.ts:77]: test should be made independent. This test depends on state from previous tests and failing when run independently.

File: src/cases/store/reducers/share-case.reducer.spec.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        case_id: '2',
+        case_fields: {
+          PersonFirstName: 'Steve',
+          PersonLastName: 'Harris',
+          PersonAddress: 'Davidson House, Forbury Square, Reading, RG1 3EB'
+        }
+      }];
+      const action = new fromActions.AddShareCases(selectedCases);
+      const state = fromReducer.shareCasesReducer(initialState, action);
+      expect(fromReducer.getShareCases(state).length).toEqual(3);
```
Comment (sreekanthpuligadda) [inline src/cases/store/reducers/share-case.reducer.spec.ts:118]: please update to make test independent.

File: src/cases/store/selectors/share-case.selectors.spec.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    spyOn(store, 'dispatch').and.callThrough();
+  });
+
+  describe('get share case state', () => {
+    it('should return search state', () => {
+      let result = [];
+      store.pipe(select(getShareCaseListState)).subscribe(value => {
+        result = value;
+      });
+      expect(result.length).toEqual(3);
```
Comment (sreekanthpuligadda) [inline src/cases/store/selectors/share-case.selectors.spec.ts:24]: please update to make test independent.
